src/wallet/wallet.cpp

```diff
@@ -1802,25 +1803,21 @@ void CWallet::ReacceptWalletTransactions()
     for (const std::pair<const int64_t, CWalletTx*>& item : mapSorted) {
         CWalletTx& wtx = *(item.second);
         CValidationState state;
-        wtx.AcceptToMemoryPool(*locked_chain, maxTxFee, state);
+        wtx.AcceptToMemoryPool(*locked_chain, state);
     }
 }
 
-bool CWalletTx::RelayWalletTransaction(interfaces::Chain::Lock& locked_chain, CConnman* connman)
+bool CWalletTx::RelayWalletTransaction(interfaces::Chain::Lock& locked_chain)
 {
     assert(pwallet->GetBroadcastTransactions());
     if (!IsCoinBase() && !isAbandoned() && GetDepthInMainChain(locked_chain) == 0)
     {
         CValidationState state;
         /* GetDepthInMainChain already catches known conflicts. */
-        if (InMempool() || AcceptToMemoryPool(locked_chain, maxTxFee, state)) {
+        if (InMempool() || AcceptToMemoryPool(locked_chain, state)) {
```

Comment:
My previous comment wasn't clear. I think this would be even better as:

```
if ((InMempool() || AcceptToMemoryPool(locked_chain, state) && pwallet->chain().p2pEnabled()) {
    pwallet->WalletLogPrintf("Relaying wtx %s\n", GetHash().ToString());
    return True;
}
```

I think the existing behaviour, where `Relaying wtx...` is logged even when p2p is not enabled and the tx is not broadcast, is a logging bug.

I can see this either way though. Changing this is a (minor) behaviour change in a PR which is supposed to be refactor only.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15288#discussion_r256036330

> I think this would be even better as

Applied your suggestion in commit "Remove use of g_connman / PushInventory in wallet code" (ae6b51043e46e80c0fe89628776de48274d7c1b2)

I think it's still not a change in behavior, unless it really is possible for `g_connman` to be null at this point. Even if there is a change, only effect is dropping a misleading log message like you say.

Reply:
Your parens are slightly different from the existing precedence.

What you have in this PR:

```
        if (InMempool() || (AcceptToMemoryPool(locked_chain, state) && pwallet->chain().p2pEnabled())) {
```

instead of:

```
        if ((InMempool() || AcceptToMemoryPool(locked_chain, state)) && pwallet->chain().p2pEnabled()) {
```

I suggest we revert the most recent change and leave tidying up for a follow-up PR. It doesn't make sense to burden this already large PR with aesthetic improvements.

Reply:
https://github.com/bitcoin/bitcoin/pull/15288#discussion_r260030584

> Your parens are slightly different from the existing precedence

Wow, not good. Reverted.